modules/engine/src/testing/index.spec.ts

```diff
@@ -90,18 +126,715 @@ describe("VectorEngine", () => {
   });
 
   // NOTE: if any of these change we have broken the rpc interface!
-  describe.skip("should properly parse rpc request schema", () => {
-    it("should fail if it has no request.id", () => {});
-    it("should fail if it has invalid request.id", () => {});
-    it("should fail if it has no request.jsonrpc", () => {});
-    it("should fail if it has invalid request.jsonrpc", () => {});
-    it("should fail if it has no request.method", () => {});
-    it("should fail if it has invalid request.method", () => {});
-    describe("should fail if it has invalid request.params", () => {
-      // TODO: all channel method RPC validation params should be tested
-      // for `invalid` fields or nonexistent fields. See vector.spec.ts
-      // or `validate.spec.ts::describe("should properly validate update
-      // schema")` for examples of types of tests + suggested test structure
+  describe("should properly parse rpc request schema", async () => {
+    const tests = [
+      {
+        name: "no request.id",
+        overrides: { id: 1, jsonrpc: "2.0", method: "chan_getConfig", params: undefined },
+        error: "should have required property 'id'",
+      },
+      {
+        name: "malformed request.id",
+        overrides: { id: -1, jsonrpc: "2.0", method: "chan_getConfig", params: undefined },
+        error: "should be >= 1",
+      },
+      {
+        name: "request.id is not a number",
+        overrides: { id: "ab", jsonrpc: "2.0", method: "chan_getConfig", params: undefined },
+        error: "should be number",
+      },
+      {
+        name: "no request.jsonrpc",
+        overrides: { id: 1, jsonrpc: undefined, method: "chan_getConfig", params: undefined },
+        error: "should have required property 'jsonrpc'",
+      },
+      {
+        name: "malformed request.jsonrpc",
+        overrides: { id: 1, jsonrpc: "3.0", method: "chan_getConfig", params: undefined },
+        error: "should be equal to one of the allowed values",
+      },
+      {
+        name: "no request.method",
+        overrides: { id: 1, jsonrpc: "2.0", method: undefined, params: undefined },
+        error: "should have required property 'method'",
+      },
+      {
+        name: "malformed request.method",
+        overrides: { id: 1, jsonrpc: "2.0", method: "invalid_method", params: undefined },
```

Comment:
hmmm, i dont think this should succeed if the `params` can be `undefined`. i believe it should be an empty body cc @rhlsthrm

Reply:
Some of the RPC schema is actually undefined. Maybe we should change it to be empty body only, I'm okay with that.

Reply:
Done